src/opts/container.rs

```diff
@@ -547,6 +547,26 @@ impl ContainerCreateOptsBuilder {
             params: self.params.clone(),
         }
     }
+
+    impl_str_field!(
+    /// The hostname to use for the container, as a valid RFC 1123 hostname.
+        hostname => "Hostname"
+    );
+
+    impl_str_field!(
+    /// The domain name to use for the container.
+        domainname => "Domainname"
+    );
+
+    impl_str_field!(
+    /// IPC sharing mode for the container. Default is "private" or "shareable", depending on daemon version.
+        ipc => "HostConfig.IpcMode"
+    );
```

Comment:
Looking at the API reference https://docs.docker.com/engine/api/v1.42/#tag/Container/operation/ContainerCreate seems like this field can be better modeled as an enum. Something like:
```
pub enum IpcMode {
    None,
    Private,
    Shareable,
    Container(String),
    Host
}
```

And then it could be added using:
```
impl_str_enum_field!(
    ipc: IpcMode => "HostConfig.IpcMode"
)
```

One thing to keep in mind is that the enum has to implement the `ToString` trait in order for this to work. Something like here: https://github.com/vv9k/docker-api-rs/blob/4cd46d2234a43523942c98e1823a4aa51cb0e1c0/src/opts/image.rs#L364-L385

Reply:
Sure, I'll take a look at the other macro implementations. Should I make a PR on [containers_api](https://github.com/vv9k/containers-api)? I noticed that's where the macros are located, and I believe podman_api would need this too

Reply:
I just saw `impl_str_enum_field` already exists

Reply:
Anyway, is there any specific location I should put the enum?